dexter/src/main/java/com/karumi/dexter/listener/threaddecorator/ThreadSpec.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (C) 2015 Karumi.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.karumi.dexter.listener.threaddecorator;
+
+/**
+ * A thread specification to execute passed runnable objects in a certain thread
+ */
+public interface ThreadSpec {
```

Comment:
@Serchinastico I think creating an abstraction is worth because:

1. As we talked with @pedrovgs on another comment, we need the `loop()` method in the `ThreadSpec` api. `Handler` class does not have the `loop` API
2. It's easier to test than a plain `Handler`
3. As I see it, the `ThreadSpec`represent the concept of running a piece of code in a given thread better than the `Handler` class does. 
4. We don't couple the thread specification to the use of a handler. In instance, if we decide to create in the future a new thread specification, then it will be a matter of just adding a new spec that may or may not need to use a `Handler`
5. We don't expose `Handler` methods that are not needed by `ThreadSpec` clients and we can add needed methods such as `loop`

Anyway if you don't agree with me, I'll be happy to discuss a better solution.